src/Emmet/Attribute.purs

```diff
@@ -0,0 +1,129 @@
+module Emmet.Attribute where
+
+import Prelude
+import DOM.HTML.Indexed.InputType (InputType, InputType(..), renderInputType) as IT
+import Data.Foldable (foldl)
+import Data.List (List, snoc)
+import Data.Maybe (Maybe(..))
+import Data.Monoid (mempty)
+import Data.Newtype (class Newtype, unwrap)
+import Data.Tuple (Tuple(..))
+
+data Attribute
+  = Class String
+  | Id String
+  | TypeInputType InputType
+  | StringAttribute String String
+  | TextContent String
+
+instance attributeEq :: Eq Attribute where
```

Comment:
You should be able to derive this instance with `derive instance eqAttribute :: Eq Attribute`